### PR TITLE
chore(ci): diff based linting

### PR DESCRIPTION
**Pull request by mikaeld**
# Description
This PR brought to you by SRE **Half Day Hack Day**. It adds the following linters to CI:

* black: https://black.readthedocs.io/en/stable/
* ruff: https://beta.ruff.rs/docs/
  * implements `pycodestyle`, `isort`, `flake8`, and others (see `pyproject.toml` for full list

## Diff based linting
Diff based linting was chosen as a way to mitigate the tremendous and immediate pain of running linters against a repo that was created 7 years ago, had tens of contributors throughout the years and suffered through the Great Python 2 ➡️ 3 Migration. It's not a perfect solution but it will help gradually improving the overall maintainability and readability.

## Helpful `make` command
The following command will run linters against files you modified.
```shell
make fixes
```

# Meme

image

### CODE REVIEW

**Comment by whd** [comment]
I vote we lint everything :trollface: 

I don't think `isort` or `black` are going to be contentious. I think we recently enabled isort in bqetl: https://github.com/mozilla/bigquery-etl/blob/main/.circleci/config.yml#L55 and black was already used. I hadn't heard of `ruff` but if it's at parity with other flake8 implementations and fast it's probably not particularly contentious to enable that either.

File: .circleci/config.yml [hunk 1]
```diff
@@ -30,6 +29,44 @@ jobs:
       - store_test_results:
           path: test-results
 
+  black:
```
Comment (haroldwoo) [inline .circleci/config.yml:32]: is this pinned to a version

File: requirements.in [hunk 2]
```diff
@@ -10,10 +10,14 @@ apache-airflow-providers-http
 apache-airflow-providers-slack
 airflow-provider-fivetran==1.1.2
 
+# Code quality
+pytest==6.2.5
+pytest-mock==3.10.0
+black==22.6.0
```
Comment (haroldwoo) [inline requirements.in:16]: nvm i see it here

**Review by haroldwoo — APPROVED** [review]
lgtm assuming most changes are resulted from running the actual linters

**Pull request merged**